### PR TITLE
docs: add java client set up guide

### PR DESCRIPTION
**Pull request by sqin2019**
#277 
1. java client does not support custom processors, discussed with Chen that we will support that when requested.
2.  used md format to format the files touched.

### CODE REVIEW

File: docs/clients.md [hunk 1]
```diff
@@ -108,4 +109,23 @@ if err != nil {
...
+
+// For gRPC services, use AuditLoggingServerInterceptor.
+AuditLoggingServerInterceptor interceptor =
+        injector.getInstance(AuditLoggingServerInterceptor.class);
+
+// Otherwise, use LoggingClient directly.
+LoggingClient client = injector.getInstance(LoggingClient.class);
+```
+
+### Extend
```
Comment (yolocs) [inline docs/clients.md:129]: Let's have the same subsections as Go. But under "### "Assemble" a client in code" we say what you say at the top: "It is not supported..."

File: docs/clients.md [hunk 2]
```diff
@@ -108,4 +109,23 @@ if err != nil {
 
 ## Java
 
-TODO
+It is not recommended and supported to "assemble" a client in code,
```
Comment (yolocs) [inline docs/clients.md:112]:
```suggestion
Currently we don't support "assembling" a client in Java code. Please file an issue if you require such capability.
```

Comment (yolocs) [reply docs/clients.md:112]: And move to the other subsection

File: docs/clients.md [hunk 3]
```diff
@@ -108,4 +109,23 @@ if err != nil {
...
+please use config file. The default config path is `audit_logging.yml`, or
+if it is overwritten via env var, AUDIT_CLIENT_CONFIG_PATH.
```
Comment (yolocs) [inline docs/clients.md:114]:
```suggestion
The default config path is `audit_logging.yml`. You can override it with env var `AUDIT_CLIENT_CONFIG_PATH`.
```

Comment (yolocs) [reply docs/clients.md:114]: Move this to "### Create a client from a config file"

File: docs/clients.md [hunk 4]
```diff
@@ -108,4 +109,23 @@ if err != nil {
...
+
+### Create a client from a config file
+
+```java
```
Comment (yolocs) [inline docs/clients.md:118]: Should we provide a intro to Guice, e.g. a link?

File: docs/clients.md [hunk 5]
```diff
@@ -108,4 +109,23 @@ if err != nil {
...
+AuditLoggingServerInterceptor interceptor =
+        injector.getInstance(AuditLoggingServerInterceptor.class);
+
+// Otherwise, use LoggingClient directly.
+LoggingClient client = injector.getInstance(LoggingClient.class);
+```
+
+### Extend
+
+Extending with custom processors is not supported at the moment, please file an issue if needed.
```
Comment (yolocs) [inline docs/clients.md:131]:
```suggestion
Due to all the log processors are initialized via injection, we currently don't support vending custom log processors to the audit log client. Please file an issue if you need such capability.
```

**Review by yolocs — CHANGES_REQUESTED** [review]
Some nits

**Review by sethvargo — APPROVED** [review]

**Review by yolocs — APPROVED** [review]

**Pull request merged**